src-electron/db/query-package.js

```diff
@@ -690,22 +690,46 @@ async function getAllPackages(db) {
     )
     .then((rows) => rows.map(dbMapping.map.package))
 }
-
+/**
+ * Retrieves attribute access interface options from the database.
+ * This function executes a query that combines results from the PACKAGE_OPTION and ATTRIBUTE tables,
+ * filtering by a specific code. It uses UNION to ensure no duplicate rows are returned, even if the same
+ * code exists in both tables. The results are then mapped to a standard format using dbMapping.map.options.
+ *
+ * @param {Object} db - The database connection object.
+ * @param {string} code - The code used to filter the results from both PACKAGE_OPTION and ATTRIBUTE tables.
+ * @returns {Promise<Array>} A promise that resolves to an array of objects, each representing an option
+ *                           with properties: PACKAGE_REF, OPTION_CATEGORY, OPTION_CODE, and OPTION_LABEL.
+ */
 async function getAttributeAccessInterface(db, code) {
+  const extendedQuery = `
+    SELECT
+        po.PACKAGE_REF,
+        po.OPTION_CATEGORY,
+        po.OPTION_CODE,
+        po.OPTION_LABEL
+    FROM
+        PACKAGE_OPTION po
+    WHERE
+        po.OPTION_CODE = ?
+
+    UNION 
+
+    SELECT
+        a.PACKAGE_REF,
+        c.NAME AS OPTION_CATEGORY,
+        a.STORAGE_POLICY AS OPTION_CODE,
+        a.NAME AS OPTION_LABEL
+    FROM
+        ATTRIBUTE a
+    LEFT JOIN CLUSTER c ON a.CLUSTER_REF = c.CLUSTER_ID
+    WHERE
+        a.STORAGE_POLICY = ?
```

Comment:
What happens if you just do the following with no union?

`SELECT
         a.PACKAGE_REF,
         c.NAME AS OPTION_CATEGORY,
         a.STORAGE_POLICY AS OPTION_CODE,
         a.NAME AS OPTION_LABEL
     FROM
         ATTRIBUTE a
     LEFT JOIN CLUSTER c ON a.CLUSTER_REF = c.CLUSTER_ID
     WHERE
         a.STORAGE_POLICY = ?`

Reply:
that's only querying the ATTRIBUTE table so it won't work

Reply:
What will it miss and how won't that work?

Reply:
The PACKAGE_OPTION table and all the global attributes

Reply:
I see a couple of issues here:
- You are retrieving all the package option entries which have accessAttributeInterface. What happens when there are more than 1 Matter SDKs loaded? Will you be pulling in package option entries from another sdk which you are not concerned with?
- You can retrieve global attributes by checking if their cluster_ref is null

Reply:
No I can't get the Cluster if it's null

Reply:
Good point about the second SDK I'll check on that